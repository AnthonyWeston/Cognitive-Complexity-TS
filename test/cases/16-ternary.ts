function f() {
    const x = true ? 1 : 0;
}

function g() {
    const x = true
        ? g()
        : g();
}

function h() {
    const x = true
        ? // depth 1
            h() // +1 +1
        : // depth 1
            false // +1 +1
            ? // depth 2
                h() // +1 +2
            : // depth 2
                h(); // +1 +2
}
