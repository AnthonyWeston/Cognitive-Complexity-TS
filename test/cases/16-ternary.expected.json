{
    "12-if-else.ts": {
        "score": 8,
        "inner": [
            {
                "name": "f",
                "score": 1,
                "line": 1,
                "column": 1
            },
            {
                "name": "g",
                "score": 5,
                "line": 5,
                "column": 1
            },
            {
                "name": "h",
                "score": 11,
                "line": 11,
                "column": 1
            }
        ]
    }
}